src/parser/sparqlParser/SparqlQleverVisitor.cpp

```diff
@@ -617,8 +601,7 @@ ParsedQuery Visitor::visit(Parser::SelectQueryContext* ctx) {
   if (!ctx->datasetClause().empty()) {
     // TODO: see if it is possible to extend reportError s.t. it can also take
     //  vector<ParserRuleContext>.
```

Comment:
We can discuss in a meeting whether it is worth it. 
If it's only for the BASE and FROM clauses, then I think the error message is already clear enough if you only specify the first one in the error message. 
In that case we could probably remove the TODO and make it an ordinary comment (or nothing)